src/DesktopIntegration.vala

```diff
@@ -43,6 +48,17 @@ public class Gala.DesktopIntegration : GLib.Object {
         // TODO: figure out if there's a better way to handle ws rearrangement
         display.window_created.connect ((window) => {
             window.workspace_changed.connect (() => windows_changed ());
```

Comment:
One more thing, this seems to be duplicate with the `workspace_changed.connect` where we also set the time_appeared_on_workspace. Or is there a reason for that?

Reply:
Fixed. Forgot to remove that